### DISCUSSION TITLE
Legend of the Red Dragon works or doesn't work depending on what optional dependencies are installed

### DISCUSSION THREAD

**Discussion by jordemort** [dosemu2/dosemu2#2104] [General]
I'm building a container with dosemu2, to run old DOS BBS door games. I'm starting with Legend of the Red Dragon.

Everything had been working well, until I decided to slim down my container image. I switched from:

```
apt-get install -y dosemu2
```

...to:

```
apt-get install -y --no-install-recommends dosemu2 install-freedos
```

This shaves about 200 meg off the size of the container image, but LORD no longer works. After attempting to detect the terminal, it dies with:

```
  ** ABNORMAL EXIT **

  ERROR : 201 Range check error
Error Address: 4DF6:2B90

  RETURNING TO THE MUNDANE WORLD...
```

Either way, I'm running dosemu2 in console mode; I don't think I should need all of the additional dependencies. Given that, It's surprising to me that not having the additional dependencies installed breaks running LORD.

Here is the diff with the packages that aren't installed on the smaller image vs the larger image: https://gist.github.com/jordemort/b1589110ce79481572f35c9d3599ee2f

The only thing on the list that seemed obvious to me is libslang2, which I think I would need for running in text mode. Even with that installed, though, the game still crashes in the same way.

The additional dependencies add up to quite a few extra packages and I'm not sure I'm prepared to bisect each one of them to figure out which one is making the game work. Any hints on which package might be causing this difference in behavior? I'm happy to gather whatever debugging information folks might want to investigate the crash, if need be.

**Comment by stsp** [comment]
After running configure, compare the
plugin list that it prints.

**Comment by stsp** [comment]
Or if you are building it not within this
container, then you need to run ldd on
libplugon_term.so to make sure no
unresolved deps.

**Reply by jordemort** [reply]
I am not building it, I am installing it from the PPA; container sources are here: https://github.com/jordemort/doorman-dosemu

**Reply by jordemort** [reply]
That is to say, the binary installed from the PPA seems to behave differently depending on what other packages are installed. If I install it with its full set of recommended packages, LORD works. If I only install the things it `Depends:` on but not the `Recommends:`, LORD crashes.

**Reply by stsp** [reply]
As I said above, use `ldd libplugin_term.so`
to look for missing deps.
